dff/context_storages/context_schema.py

```diff
@@ -0,0 +1,286 @@
+                    last_keys = sorted(nest_dict.keys())[limit:]
+                    ctx_dict[field_name] = {k: v for k, v in nest_dict.items() if k in last_keys}
+                elif len(nest_dict) < field_props.subscript and last_read_key > field_props.subscript:
+                    limit = field_props.subscript - len(nest_dict)
+                    tasks[field_name] = log_reader(limit, field_name, primary_id)
+            else:
+                tasks[field_name] = log_reader(None, field_name, primary_id)
+
+        if self.supports_async:
+            tasks = dict(zip(tasks.keys(), await gather(*tasks.values())))
+        else:
+            tasks = {key: await task for key, task in tasks.items()}
+
+        for field_name in tasks.keys():
+            log_dict = {k: v for k, v in tasks[field_name].items()}
+            ctx_dict[field_name].update(log_dict)
+
+        ctx = Context.cast(ctx_dict)
+        setattr(ctx, ExtraFields.primary_id.value, primary_id)
+        setattr(ctx, ExtraFields.storage_key.value, storage_key)
+        return ctx
+
+    async def write_context(
+        self,
+        ctx: Context,
+        pac_writer: _WritePackedContextFunction,
+        log_writer: _WriteLogContextFunction,
+        storage_key: str,
+        chunk_size: Union[Literal[False], int] = False,
+    ):
+        """
+        Write context to storage.
+        Calculate what fields to write, split large data into chunks if needed and call writer function.
+        Also update `updated_at` attribute of the given context with current time, set `primary_id` and `storage_key`.
+
+        :param ctx: the context to store.
+        :param pac_writer: the function used for writing context to
+            `CONTEXT` table (see :py:const:`~._WritePackedContextFunction`).
+        :param log_writer: the function used for writing context to
+            `LOGS` table (see :py:const:`~._WriteLogContextFunction`).
+        :param storage_key: the key to store the context with.
+        :param chunk_size: maximum number of items that can be inserted simultaneously, False if no such limit exists.
+
+        :return: the read :py:class:`~.Context` object.
+        """
+        updated_at = time.time_ns()
+        setattr(ctx, ExtraFields.updated_at.value, updated_at)
+        created_at = getattr(ctx, ExtraFields.created_at.value, updated_at)
+
+        ctx_dict = ctx.model_dump()
+        logs_dict = dict()
+        primary_id = getattr(ctx, ExtraFields.primary_id.value, str(uuid4()))
+
+        for field_props in [value for value in dict(self).values() if isinstance(value, SchemaField)]:
+            nest_dict = ctx_dict[field_props.name]
+            last_keys = sorted(nest_dict.keys())
+
+            if (
+                self.append_single_log
+                and isinstance(field_props.subscript, int)
+                and len(nest_dict) > field_props.subscript
+            ):
+                unfit = -field_props.subscript - 1
```

Comment:
Because it represents the number of elements that didn't fit into `CONTEXTS` table. Alsp it's deep inside of the function body, so I don't really think that matters.